File: 0189-rotate-array/0189-rotate-array.py
class Solution(object):
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        k %= len(nums)  # Handle cases where k > len(nums)
        rotated_arr = nums[-k:] + nums[:-k]  # Rotate array efficiently
        nums[:] = rotated_arr
        

File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int len = nums.size();
        if(k>len){
            k = k%len;
        }
        vector<int> new_nums(len);
        if(k%len != 0 || k != 0){
            for(int i=0;i<len;i++){
                new_nums[(i+k)%len] = nums[i];
            }
            for(int i = 0; i<len;i++){
                nums[i] = new_nums[i];
            }
        }
    }
};